lms/djangoapps/verify_student/views.py

```diff
@@ -1047,3 +1055,187 @@ def reverification_window_expired(_request):
     """
     # TODO need someone to review the copy for this template
     return render_to_response("verify_student/reverification_window_expired.html")
+
+
+@login_required
+def incourse_reverification_confirmation(_request):  # pylint: disable=invalid-name
+    """
+    Shows the user a confirmation page if the submission to SoftwareSecure was successful
+    """
+    return render_to_response("verify_student/incourse_reverification_confirmation.html")
+
+
+class InCourseReverifyView(View):
+    """
+    The in-course reverification view.
+    Needs to perform these functions:
+        - take new face photo
+        - retrieve the old id photo
+        - submit these photos to photo verification service
+
+    Does not need to worry about pricing
+    """
+    @method_decorator(login_required)
+    def get(self, request, course_id, checkpoint_name):
+        """
+        display this view
+        """
+        user = request.user
+        course_key = CourseKey.from_string(course_id)
+        course = modulestore().get_course(course_key)
+        if course is None:
+            raise Http404
+        checkpoint = VerificationCheckpoint.get_verification_checkpoint(course_key, checkpoint_name)
+        if checkpoint is None:
+            raise Http404
+        init_verification = SoftwareSecurePhotoVerification.get_initial_verification(user)
+        if not init_verification:
+            return redirect(reverse('verify_student_verify_later', kwargs={'course_id': unicode(course_key)}))
+
+        display_steps = self._display_steps()
+        requirements = self._requirements(display_steps, request.user.is_active)
```

Comment:
Unlike the full pay/verify flow, this page does not require multiple steps.  We basically want to show a single page that allows users to take a photo of their face.  For this reason, we don't need to provide display steps or requirements.

It looks like you're using the top-level Backbone views from the payment and verification flow.  I'd recommend re-using just the views that render the webcam.  You can define new top-level views to construct the rest of the page (which will be much simpler than the payment/verification flow, since it's only one page).